src/metaopt/io/database/mongodb.py

```diff
@@ -0,0 +1,137 @@
+import json
+import bson
+import pymongo
+
+from pymongo import MongoClient
+from datetime import datetime
+from abstractdb import AbstractDB
+
+try:
+    from StringIO import StringIO
+except ImportError:
+    from io import StringIO
+
+try:
+    # Python 3.x
+    from urllib.parse import quote_plus
+except ImportError:
+    # Python 2.x
+    from urllib import quote_plus
+
+
+class MongoDB(AbstractDB):
+    def __init__(self, dbhost='localhost', dbname='metaopt', dbport=27017, \
+                                            username=None, password=None):
+        self.dbhost = str(dbhost)
+        self.dbname = str(dbname)
+        self.username = str(username)
+        self.password = str(password)
+
+        # we might have a different port for the database port
+        # so here we do a check if the port is there
+
+        self.dbport = str(dbport)
+
+        # set the connection string
+        # this should be mongodb://mongodb0.example.net:27017
+        # according to MongoDB documentation
+        # https://docs.mongodb.com/getting-started/python/client/
+
+        self.uri = 'mongodb://' + self.dbhost + ':' + self.dbport
```

Comment:
Building the DB url (you typed uri by the way, was it intentional?) can get complicated if we support many options. That would deserve it's own method. It could be worth making it a dynamic `@property`.

Reply:
I used the full URI until our discussion last week with Fred that for current localhost testing purposes we shouldn't put a default username and password because it would get passed on. Anyway, I'll comment out this URI and use the one following this line instead.

Reply:
I think that the `@property` is a good idea. I am not sure how we should ask for the password. There are multiple ways of authentications here: https://api.mongodb.com/python/current/examples/authentication.html

Also, there this useful stl in 2.7, >=3.5: https://docs.python.org/3/library/getpass.html